modules/FIC_Github.py

```diff
@@ -26,6 +28,33 @@ def _init_github(self, *args):
         self.repo_data = github3.GitHub.repository(args[0], args[1], args[2])
         return self.repo_data
 
+    def limit_checker(self):
+        limit_requests = self._gh.ratelimit_remaining
+        self.LOGGER.info("Rate limit is: %s", limit_requests)
+
+        if limit_requests < 5:
+            return self.switch_token()
+
+        else:
+            return True
+
+    def switch_token(self):
+        self.get_token()
+        self.__init__()
```

Comment:
Is `self.__init__()` needed? Seems a bit weird.

Reply:
yes, it is needed. Any time the token is changed the program required a re-logging, more exactly it must reinitialize the following attributes:
```python
self._token = os.environ.get(GIT_TOKEN) # the new token
self._gh = self._auth() # github3.login(token=self._token) ->logging with new token
```
If it is necessary,  I think that we can replace it adding a new method to do this ^ (the re-logging part using the new token)

Reply:
Implemented the re-logging part in _switch_token(self) method to avoid the call of def __init__(self):
Commit -> https://github.com/mozilla-releng/firefox-infra-changelog/pull/416/commits/179b576fa3203e494c6136ee6b735041f82bc792#diff-81d9b860925058fdb72172aa8f828962R71